appveyor/Convert-FileUrisToGitHubUris.ps1

```diff
@@ -0,0 +1,120 @@
+}
+
+function Get-CaseSensitivePath($pathName)
+{
+  $pathExists = Test-Path($pathName)
+  if (-Not $pathExists)
+  {
+      return $pathName
+  }
+  $directoryInfo = New-Object IO.DirectoryInfo($pathName)
+
+  if ($directoryInfo.Parent -ne $null)
+  {
+      $parentPath = Get-CaseSensitivePath($directoryInfo.Parent.FullName)
+      $childPath = $directoryInfo.Parent.GetFileSystemInfos($directoryInfo.Name)[0].Name
+      return(Join-Path $parentPath $childpath -resolv)
+  }
+  else
+  {
+      return $directoryInfo.Name
+  }
+}
+
+$sarifLog = (Get-Content $location) -join "`n" | ConvertFrom-Json
+
+for ($i = 0; $i -lt $sarifLog.runs.Count; $i++)
+{
+  #files
+  for($j = 0; $j -lt $sarifLog.runs[$i].files.Count; $j++)
+  {
+    $sarifLog.runs[$i].files[$j].uri = Rebase-Uri($sarifLog.runs[$i].files[$j].uri)
+    #UNTESTED
+    $props = JSON.runs[$i].files[$j].PSObject.Properties
+    if($props.Name)
+    {
+      $URIReplacement = $props.Name
+      $content = $props.Value
+      $sarifLog.runs.files.PSObject.Properties.Remove($props.Name)
+      $sarifLog.runs.files | add-member -Name Rebase-Uri($URIReplacement) -Value $content -MemberType NoteProperty
+    }
+  }
+
+  #results
+  for($j = 0; $j -lt $sarifLog.runs[$i].results.Count; $j++)
+  {
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].locations.Count; $k++)
+    {
+      $location = $sarifLog.runs[$i].results[$j].locations[$k]
+      if(!$location.resultFile)
+      {
+        $sarifLog.runs[$i].results[$j].locations[$k].resultFile.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].locations[$k].resultFile.uri)
+      }
+
+      if(!$location.analysisTarget)
+      {
+        $sarifLog.runs[$i].results[$j].locations[$k].analysisTarget[$l].uri = Rebase-Uri($sarifLog.runs[$i].results[$j].locations[$k].analysisTarget[$l].uri)
+      }
+
+    }
+
+    #Codeflow
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].codeFlows.Count; $k++)
+    {
+      for($l = 0; $l -lt $sarifLog.runs[$i].results[$j].codeFlows[$k].locations.Count; $l++)
+      {
+        $sarifLog.runs[$i].results[$j].codeFlows[$k].locations[$l].physicalLocation.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].codeFlows[$k].locations[$l].physicalLocation.uri)
+      }
+    }
+
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].relatedLocations.Count; $k++)
+    {
+      $sarifLog.runs[$i].results[$j].relatedLocations[$k].physicalLocation.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].relatedLocations[$k].physicalLocation.uri)
+    }
+
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].stacks.Count; $k++)
+    {
+      $sarifLog.runs[$i].results[$j].stacks[$k].frames.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].stacks[$k].frames.uri)
+    }
+
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].fixes.Count; $k++)
+    {
+      $sarifLog.runs[$i].results[$j].fixes[$k].fileChanges.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].fixes[$k].fileChanges.uri)
```

Comment:
`fileChanges` is itself an array: a single "fix" might require you to modify one or more files. Each changed file is represented by a `fileChange` object. The `fileChange` object, in turn, consists of a `uri` saying which file is being changed, together with an array of `replacement` objects; each `replacement` object specifies a single contiguous range of bytes to modify within the file.

So you'll need a nested loop over the elements of the `fileChanges` array.